modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/HttpMethodFeature.java

```diff
@@ -155,14 +156,15 @@ private Method _getSuperMethod(Method method) {
 		HttpMethodFeature.class);
 
 	private BundleContext _bundleContext;
+
+	@Reference(target = "(oauth2.scope.checker.type=http.method)")
+	private HttpMethodScopeLogic _httpMethodScopeLogic;
```

Comment:
What about updating also the reference here to the specific ScopeLogic `AnnotationScopeLogic`?

https://github.com/liferay-objects/liferay-portal/blob/cd4114cb38b9d65fd3cda239a93d56c13a959e7e/modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/AnnotationFeature.java#L106

Reply:
We would need to publish a `AnnotationScopeLogic` service in that case. I thought because AnnotationFeature didn't need anything beyond the exported `ScopeLogic` interface's method signature, it would be more clear if it uses that interface. Do you see a benefit?

Reply:
Ok agree, I didn't notice that you were using the public method in the HttpMethodScopeLogic with the `String requestMethod` parameter, that makes sense